src/transformStateWithClones.js

```diff
@@ -7,7 +7,37 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateArray = [];
+
+  let stateCopy = {
+    ...state,
+  };
+
+  for (const part of actions) {
+    switch (part.type) {
+      case 'addProperties':
+        stateCopy = {
+          ...stateCopy,
+          ...part.extraData,
+        };
+        stateArray.push({ ...stateCopy });
+        break;
+
+      case 'removeProperties':
+        for (const removeElement of part.keysToRemove) {
+          delete stateCopy[removeElement];
+        }
+        stateArray.push({ ...stateCopy });
+        break;
+
+      case 'clear':
+        stateCopy = {};
+        stateArray.push({});
+        break;
+    }
```

Comment:
![image](https://user-images.githubusercontent.com/96409563/216539756-2985ae63-173f-46a3-a0bb-8e14d63a884a.png)
Also, you can push the stateCopy after all cases in order not to duplicate yourself